### PR TITLE
Fix generated comments to reflect azure-service-operator

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
**What this PR does / why we need it**:

Our generated code had comments indicating the source was `k8s-infra` and `k8s-infra-gen` - we change these to read `azure-service-operator` while retaining the ability to detect and clean up generated code from older versions.

**How does this PR make you feel**:
gif

### CODE REVIEW

File: hack/generator/pkg/astmodel/generation_consts.go [hunk 1]
```diff
@@ -8,7 +8,14 @@ package astmodel
 var (
 	CodeGenerationComments []string = []string{
 		// Note that this format is actually an official specification in go:  https://github.com/golang/go/issues/13560
-		"// Code generated by k8s-infra. DO NOT EDIT.", // TODO: Update this when the generated is moved/renamed
+		"// Code generated by azure-service-operator. DO NOT EDIT.", // TODO: Update this when the generated is moved/renamed
```
Comment (babbageclunk) [inline hack/generator/pkg/astmodel/generation_consts.go:11]: Maybe remove the TODO? Although I guess we still want to update it if it's moved again.

File: hack/generator/pkg/astmodel/generation_consts.go [hunk 2]
```diff
@@ -8,7 +8,14 @@ package astmodel
...
+	}
+
+	// Required for backward compatibility.
+	// Presence of any one string will result in the file being treated as a generated file that we are responsible for
+	CodeGenerationCommentsDeprecated []string = []string{
```
Comment (matthchr) [inline hack/generator/pkg/astmodel/generation_consts.go:16]: Is the plan to delete this later?

Feels like we should maybe just delete this now and tell the 3 or 4 of us to manually clean up the directory once when we pull? Cleaner than having legacy here before we're even released?

**Review by matthchr — APPROVED** [review]

**Review by babbageclunk — APPROVED** [review]

File: hack/generator/pkg/astmodel/generation_consts.go (hunk 1 above)
Comment (matthchr) [reply hack/generator/pkg/astmodel/generation_consts.go:11]: I agree - remove TODO. I think the TODO was in reference to what @theunrepentantgeek just did (i.e. that it shouldn't say k8sinfra)

Comment (matthchr) [inline hack/generator/pkg/astmodel/generation_consts.go:11]: IMO we should differentiate between ASO and the tool that generates aso.

Maybe this is ASO-gen or something?

Comment (theunrepentantgeek) [reply hack/generator/pkg/astmodel/generation_consts.go:11]: Nah, that TODO has been there for ages. I'll remove it though.

File: hack/generator/pkg/astmodel/generation_consts.go (hunk 2 above)
Comment (theunrepentantgeek) [reply hack/generator/pkg/astmodel/generation_consts.go:16]: Yeah, didn't take a lot to persuade me. :grin:

File: hack/generator/pkg/astmodel/generation_consts.go (hunk 1 above)
Comment (theunrepentantgeek) [reply hack/generator/pkg/astmodel/generation_consts.go]: Changed to `azure-service-operator-codegen`.

**Pull request merged**